ha/core/config/consul_kv_store.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import consul
+import errno
+import socket
+from urllib.parse import urlparse
+
+class ConsulKvStore:
+    """ Represents a Consul kv Store """
+
+    name = "consul"
+    version = "/v1"
+
+    def __init__(self, url: str):
```

Comment:
This is Consul kv store only, so why is prefix required in url. There will never be any other prefix than "consul".
It will be better to have signature like this -
def __init__(self, prefix, host='localhost', port=8500):
       pass

Reply:
Done modified to 
```
def init(self, prefix, host='localhost', port=8500):
```